streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -798,6 +806,12 @@ void runOnce() {
                         final int committed = taskManager.maybeCommitActiveTasksPerUserRequested();
 
                         if (committed > 0) {
+                            if (eosBetaEnabled) {
+                                taskManager.commitAllActive();
```

Comment:
prop: add a comment here for reasoning of committing all tasks at once

Reply:
I am not a fan of too many comments -- we test for `eosBetaEnabled` and one just need to know what "eos beta" is and what it implies...

Reply:
I agree, it's just not super intuitive why we would commit for all in this case. Feel free